### PR TITLE
Add class book, classroom, and rentals with their associations

### PR DESCRIPTION
**Pull request by UbdaNam**
## What does this PR do?

This PR is raised to implement the following features: 👍 :

- [x] Use GitFlow to introduce new features
- [x] Add classes book, classroom, and rentals
- [x] Add associations to existing classes using UML diagram

## Description of the tasks completed 👇 

- [x] Add classes book, classroom, and rentals with their associations
- [x] Install ruby
- [x] Use the linter tool that analyzes my source code to flag programming errors, bugs, stylistic errors, and suspicious constructs. 💯 
  
## How should it be tested

Clone the repository
```
git clone https://github.com/UbdaNam/OOP-school-library.git
```
Navigate to the folder
```
cd OOP-school-library
```
Checkout the branch
```
git checkout feature_association
```

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -1,11 +1,18 @@
 require './person'
```
Comment (rloterh) [inline student.rb:1]: - [ ] [OPTIONAL]
By convention, it's a good practice to use `require_relative` classes in the same source and use `require` for external modules or packages. Kindly ensure to improve upon this aspect.

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Book
+  attr_accessor :title, :author
+  attr_reader :rentals
```
Comment (rloterh) [inline book.rb:3]: - [ ] `rental` should be marked with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ it. Kindly ensure to improve upon this aspect.

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+  def initialize(_title, author)
+    @title = label
+    @author = author
+    @rentals = []
+  end
```
Comment (rloterh) [inline book.rb:9]: - [ ] As required, let's ensure that a method for adding rental is present. The _book_ itself should be sent as a parameter to create the new _rental_, thereby achieving the _'has-many'_ association. We can achieve this by modifying our code to look like the following example below:

```ruby
class Book
  attr_accessor :title, :author, :rentals
  .
  .
  def add_rental(person, date)
    Rental.new(date, self, person)
  end
end
```

File: rental.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Rental
+  attr_reader :book, :person
```
Comment (rloterh) [inline rental.rb:2]: - [ ] Let's mark `book` and `person` with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ to it. Kindly ensure to improve upon this aspect.

File: person.rb [hunk 5]
```diff
@@ -4,14 +4,15 @@
 
 class Person < Nameable
   attr_accessor :name, :age
-  attr_reader :id
+  attr_reader :id, :rentals
```
Comment (rloterh) [inline person.rb:7]: - [ ] Here again `rental` should be marked with `attr_accessor`, as we intend to both _access (get)_ as well as _write (set)_ it. Kindly ensure to improve upon this aspect.

File: person.rb [hunk 6]
```diff
@@ -4,14 +4,15 @@
...
 
   def initialize(age, name = 'Unknown', parent_permission: true)
     super()
     @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @rentals = []
   end
```
Comment (rloterh) [inline person.rb:17]: - [ ] Let's ensure a method for adding _rentals_ is present, so as to establish a a _rental_ `belong-to` _person_ relationship, like the code example below:

```ruby
class Person
  attr_accessor :name, :age, :rentals
  .
  .

  def add_rental(book, date)
    Rental.new(date, book, self)
  end

  .
  .
end
```

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Abdurahim, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there!
🏁 🚦 ⛪ 🚜 🏃 🌴 🏫 🌳 🏘️

## Highlights 👍  
- Good job so far with creating associations among objects!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by UbdaNam** [comment]
Thank you @rloterh 🥇 for the required changes and your explanation. I have applied the changes requested. I will request rereview 👍

**Review by laiifuu — APPROVED** [review]
# STATUS: APPROVED :heavy_check_mark: 🎉 

## Hi @UbdaNam :wave: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

friends-joey

## Highlights
- [x] Implemented the required changes :heavy_check_mark: 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**